Makefile

```diff
@@ -111,6 +111,25 @@ lint:
 	@echo "Linter: gosimple\n--------------------"
 	@gometalinter --disable-all --deadline 45s --enable=gosimple extkeys cmd/... geth/... | grep -v -f ./static/config/linter_exclude_list.txt || echo "OK!"
 
+install:
+	go get -u github.com/golang/dep/cmd/dep
+	dep ensure
+
+	mkdir _vendor_keep
```

Comment:
Can we just leave it without `dep prune` (until it works as we want it)? If `dep prune` is broken for now, introducing this obscure dirs manipulations will introduce technical debt.

It works fine without pruning, right?

Reply:
Right, but it turned out that we will add around 1.2M LOC because of one package that contains translation files... I'd like to try a different approach and remove only those translation packages and compare numbers.

Reply:
I'm a huge fan of pruning myself - in many cases we need only a few files from dependency. But it should be completely automatic and reliable process, which `dep` at some point will be able to accomplish. So between "no pruning at all" (which takes more space, but that's still ok) and "full pruning" I'd choose second. But to "half-manual pruning using hack in Makefile" I'd prefer first one.

This script is undoubtedly doing its job now, but:
 -  it's easy to forger about (I rarely use `make` command, for example, and always assume that standard tooling `go build`/`go test`/`dep` should work).
 - it's not cross-platform (imagine we have tomorrow some strange dev on Windows) - we'll need to rewrite this script to support Windows
 - it's error prone (each line potentially can fail and we don't handle errors in this script)
 - etc/etc.

So my suggestion is to wait until your issue in dep is resolved, perhaps create technical debt issue for prune. Moreover `prune` command will be eliminated from `dep` in the next releases, which only adds complexity in maintaining this script.

Reply:
Without pruning it will add `5029 files changed, 1145282 insertions(+), 1 deletion(-)` for our fork of go-ethereum and `2631 files changed, 1018238 insertions(+)` for status-go. I suppose we can handle that as it's a one-time operation. 

This issue is blocked, for the time being, so no rush. We should wait for a comment from a team building `dep` and then decide what to do. 

Regarding your comments:
1. Most people do :) `Makefile` is understood by a broader group of developers, not only ones who are fluent in Go. Even if `dep prune` would work correctly, how many people would know that they need to run `dep ensure && dep prune` to properly set up dependencies? `make install` nicely hides this implementation detail.
2. How is it not cross-platform? 
3. If any line fails, the script will just stop. No need to handle anything.

I don't like this hack either but adding so many unnecessary files just because a tool is broken also does not seem like a great idea. That's why I wanted to see if it's doable to make pruning work with a little help of bash.

Let's get back to this when we rebase on geth 1.7.0.

Reply:
Okay. Just want to emphasize that we already have non-pruned dependencies, and most of the projects do as well. And unless deps are really huge (>100M), that amount of files doesn't really cause any harm. At least, comparing to the technical debt of a workaround.

As for the comments:
1. Using Makefile and default tools aren't mutually exclusive things. Also, `dep ensure && dep prune` would be "standard" `go dep ensure` soon :)
2. I don't think Windows is POSIX-compatible and commands like `cp -R` will work.
3. It will likely leave _vendor_keep dir.

Reply:
> I don't think Windows is POSIX-compatible and commands like cp -R will work.

I wouldn't worry about that. Without a proper shell like cygwin, one won't be able to run this Makefile anyway.

> It will likely leave _vendor_keep dir.

Correct, it would need to be removed manually.

I got a response from the team building dep and including `prune` into `ensure` won't happen soon. We're talking about months.